syllabuspdf/views.py

```diff
@@ -100,3 +100,31 @@ def index(request):
         'settings': form_settings
     }
     return render(request, 'syllabuspdf/index.html', context)
+
+@csrf_exempt
+def logger_view(request):
+    '''
+    Logger API - to log onclick event from Generate PDF button
+    '''
+    # Note we could make this more dynamic and parameterize the request
+    try:
+        course_id = None
+        user_id = None
+        if "course_id" not in request.session and "user_id" not in request.session:
+            raise Exception("course_id", "user_id")
+        if "course_id" not in request.session:
+            raise Exception("course_id", None)
+        if "user_id" not in request.session:
+            raise Exception("user_id", None)
+        course_id = request.session['course_id']
+        user_id = request.session['user_id']
+        logger.info(f"Syllabus index: Syllabus PDF Generated: course_id={course_id} user_id={user_id}")
+        return HttpResponse(status=200)
+    except Exception as e:
+        err1,err2 = e.args
+        if err1 and err2:
+            logger.error(f"Syllabus index: Syllabus PDF Not Generated: canvas {err1} and {err2} not found in session")
+            return HttpResponseNotFound(f'{err1} and {err2} not found')
+        if err1:
+            logger.error(f"Syllabus index: Syllabus PDF Not Generated: canvas {err1} not found in session")
+            return HttpResponseNotFound(f'{err1} not found')
```

Comment:
The try/except then conditional block is a bit chunky. Looks like you are checking for both session attributes, then checking for each one individually? What do you think of something like:
```py
try:
    course_id = request.session['course_id']
    user_id = request.session['user_id']
    logger.info(f"Syllabus index: Syllabus PDF Generated: course_id={course_id} user_id={user_id}")
    return HttpResponse(status=200)
except KeyError as e: ...the rest of your code
```
That would condense it a bit, and still get at the fact that both `course_id` and `user_id` should be in `request.session`

Reply:
oh i see! that looks much shorter than what I wrote, Thanks Tylor. Will look into updating the code with this.